rest/rest-resources/src/main/java/org/apache/brooklyn/rest/util/MultiSessionAttributeAdapter.java

```diff
@@ -217,7 +219,11 @@ private void invalidateAllSession(HttpSession preferredSession, HttpSession loca
         private HttpSession findValidPreferredSession(HttpSession localSession, HttpServletRequest optionalRequest) {
             if (localSession instanceof Session) {
                 SessionHandler preferredHandler = getPreferredJettyHandler((Session)localSession, true, true);
-                HttpSession preferredSession = preferredHandler==null ? null : preferredHandler.getHttpSession(localSession.getId());
+
+                SessionIdManager idManager = ((DefaultSessionIdManager)preferredHandler.getSessionIdManager());
```

Comment:
Why do we need to cast it to `DefaultSessionIdManager`?

Reply:
I see this code is mostly a copy of the impl of `sessionHandler.getHttpSession(String)`, which is now protected in jetty 9.4.22.

But your code does not call `session.setIdChanged(true)`, which that method does when the id doesn't match. I've no idea what the implications are of that difference!